util/request.py

```diff
@@ -27,22 +27,25 @@ def crossorigin(anonymous=True):
     def decorate(func):
         @wraps(func)
         def wrapper(*args, **kwargs):
-            cors_origin = app.config.get("CORS_ORIGIN", "*")
+            cors_origins = app.config.get("CORS_ORIGINS", "*")
             headers = BASE_CROSS_DOMAIN_HEADERS
 
             # For calls that can only be called from
             # a known cross-origin domain like CSRF token
             # request
-            if not anonymous and cors_origin == "*":
+            if not anonymous and cors_origins == "*":
                 return func(*args, **kwargs)
 
             credentials = False
-            if cors_origin != "*":
+
+            origin = request.environ.get("HTTP_ORIGIN")
+            if "*" not in cors_origins and origin in cors_origins:
```

Comment:
not sure I understand the need for the `HTTP_ORIGIN` environment variable

Reply:
In the current implementation we rely on the `cors_origin` to be whatever is set in `CORS_ORIGIN`, but now that `CORS_ORIGIN` could be multiple origins, we need to get the requested origin and check if it's in the list of supported `CORS_ORIGIN`